cirq/experiments/t2_decay_experiment.py

```diff
@@ -163,21 +200,77 @@ def t2_decay(
 
     y_basis_measurements = results[abs(results.inv_y) > 0]
     x_basis_measurements = results[abs(results.inv_x) > 0]
-    x_basis_tabulation = pd.crosstab(x_basis_measurements.delay_ns,
-                                     x_basis_measurements.output).reset_index()
-    y_basis_tabulation = pd.crosstab(y_basis_measurements.delay_ns,
-                                     y_basis_measurements.output).reset_index()
 
-    # If all measurements are 1 or 0, fill in the missing column with all zeros.
-    for tab in [x_basis_tabulation, y_basis_tabulation]:
-        for col_index, name in [(1, 0), (2, 1)]:
-            if name not in tab:
-                tab.insert(col_index, name, [0] * tab.shape[0])
+    if num_pulses and len(num_pulses) > 1:
+        max_pulses = max(num_pulses)
+        cols = [f'pulse_{t}' for t in range(max_pulses)]
+        x_basis_measurements['num_pulses'] = x_basis_measurements[cols].sum(
+            axis=1)
+        y_basis_measurements['num_pulses'] = y_basis_measurements[cols].sum(
+            axis=1)
+
+    x_basis_tabulation = _create_tabulation(x_basis_measurements)
+    y_basis_tabulation = _create_tabulation(y_basis_measurements)
 
     # Return the results in a container object
     return T2DecayResult(x_basis_tabulation, y_basis_tabulation)
 
 
+def _create_tabulation(measurements):
+    """Returns a sum of 0 and 1 results per index from a list of measurements.
+    """
+    if 'num_pulses' in measurements.columns:
+        cols = [measurements.delay_ns, measurements.num_pulses]
+    else:
+        cols = [measurements.delay_ns]
+    tabulation = pd.crosstab(cols, measurements.output).reset_index()
+    # If all measurements are 1 or 0, fill in the missing column with all zeros.
+    for col_index, name in [(1, 0), (2, 1)]:
+        if name not in tabulation:
+            tabulation.insert(col_index, name, [0] * tabulation.shape[0])
+    return tabulation
+
+
+def _cpmg_circuit(qubit: devices.GridQubit, delay_var: sympy.Symbol,
+                  max_pulses: int):
```

Comment:
This will also remove inconsistency in conventions for the interpretation of `delay_var` between your Hahn echo and CPMG code (in the former delay_var is *half* the time between the initial pi/2 pulse and the subsequent pi pulse, but in CPMG it is exactly the time between the pi/2 and pi).

Not sure whether we care about backwards compatibility here. If not, I think your CPMG convention is to be preferred.

Reply:
Done.  I doubt people are using this, but. if so. I have broken their timing.